external-import/recorded-future/src/rflib/rf_to_stix2.py

```diff
@@ -1104,6 +1104,14 @@ def _create_report_types(self, topics):
         return list(ret)
 
     def to_stix_objects(self):
+        report_object_refs = [obj.id for obj in self.objects]
+        # Report in STIX lib must have at least one object_refs even if there is no object_refs
+        if len(report_object_refs) == 0:
+            # Put a fake ID in the report
+            report_object_refs.append(
+                "intrusion-set--fc5ee88d-7987-4c00-991e-a863e9aa8a0e"
+            )
+
```

Comment:
I think this block should be after the method docstring and not before

I would suggest to use the author ref rather a fake intrusion-set if the report

```suggestion
        """Returns a list of STIX objects"""
       
        # Report in STIX lib must have at least one object_refs 
        report_object_refs = [obj.id for obj in self.objects]
        if len(report_object_refs) == 0:
            # Add the author if needed
            report_object_refs.append(
               self.author.id
            )

```

Reply:
The author cannot be referenced in the `object_refs` because the report does not concern them. The report was created by the author, but they are not involved in the report at any point. From a business perspective, an analyst does not want the author included in the entities they reference. Additionally, the author was already removed previously following a request from RF themselves.  

The relationship here will never be created because the ID does not exist on the platform. The workaround is to create a fake ID for the reasons I mentioned: the `object_refs` field is mandatory, but if there are no referenced objects, we still want to create the report.  

I agree with the point about the docstring needing to come earlier, and I’ll make that change.

Reply:
Then I think you can sublclass Report to make objects optional 

```
from stix2 import Report
import pycti
from collections import OrderedDict
from stix2.properties import ListProperty, ReferenceProperty


class CustomReport(Report):
    """Subclass of Report with 'object_refs' property set to required=False."""
    
    _properties = OrderedDict(Report._properties)  # Copy the parent class properties
    _properties['object_refs'] = ListProperty(
        ReferenceProperty(valid_types=["SCO", "SDO", "SRO"], spec_version='2.1'),
        required=False
    )

my_report = CustomReport(
    id=pycti.Report.generate_id(name="My Report", published="2020-01-01T00:00:00Z"),
    name="My Report",
    description="This is a custom report without objects",
```

Reply:
@flavienSindou thanks for the suggestion, works perfectly well locally, I will implement that :)